samples/matter/lock/src/main.cpp

```diff
@@ -17,3 +23,23 @@ int main()
 	LOG_ERR("Exited with code %" CHIP_ERROR_FORMAT, err.Format());
 	return err == CHIP_NO_ERROR ? EXIT_SUCCESS : EXIT_FAILURE;
 }
+
+/* Override weak k_sys_fatal_error_handler method to implement saving crash data to the retention memory */
+void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *esf)
```

Comment:
Does this work with `CONFIG_RESET_ON_FATAL_ERROR`? I can see that then a non-weak implementation from NCS is pulled in.

Reply:
Yeah, indeed it doesn't work with that config :/ I'm looking for a better solution for that in this case...

Reply:
I added an additional `NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_RESET_ON_FATAL_ERROR` config that must be added instead of `CONFIG_RESET_ON_FATAL_ERROR` when crash logs support is enabled. I added also proper information to our docs regarding that and disabled the `CONFIG_RESET_ON_FATAL_ERROR` config in the prj.conf related to diagnostic-logs snippet.

What do you think about that? Is it enough?

Reply:
Wrapped the `z_fatal_error` function and it works fine :)